### PR TITLE
Refactors DesignSystem.swift to not be singleton and updates documentation.

### PR DESCRIPTION
**Pull request by raonivaladares**
# Description

- Updates DSColor documentation to warning about .cgColor.
- Refactors DesignSystem to not be a Singleton.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: Sources/Public/DesignSystem.swift [hunk 1]
```diff
@@ -17,22 +17,19 @@
         It's necessary to configure the Design System current Brand first
         or fatalError will be raised.
 
-            DesisgnSystem.shared.configure(with: Brand)
-
- - Warning
-        This class is a singleton.
+            DesisgnSystem().configure(with: Brand)
```
Comment (viviancrs) [inline Sources/Public/DesignSystem.swift:20]: Typo on DesisgnSystem

File: Sources/Core/Configuration/ConfigurationStorable.swift [hunk 2]
```diff
@@ -0,0 +1,4 @@
+protocol ConfigurationStorable: class {
+    func save(_ theme: Theme)
```
Comment (viviancrs) [inline Sources/Core/Configuration/ConfigurationStorable.swift:2]: pra maior coerência com o getTheme aqui não deveria saveTheme?

File: Sources/Public/DesignSystem.swift (hunk 1 above)
Comment (raonivaladares) [reply Sources/Public/DesignSystem.swift:20]: Done!

File: Sources/Core/Configuration/ConfigurationStorable.swift (hunk 2 above)
Comment (raonivaladares) [reply Sources/Core/Configuration/ConfigurationStorable.swift:2]: faz sentido! 
se eu remover o "wild card" e deixar o save(theme: Theme) funciona?
ou fica melhor saveTheme(_:) mesmo?

Comment (raonivaladares) [reply Sources/Core/Configuration/ConfigurationStorable.swift:2]: Subi correção com  save(theme: Theme).

File: Tests/Public/DesignSystem+Spec.swift [hunk 3]
```diff
@@ -45,6 +47,60 @@ final class DesignSystemSpec: QuickSpec {
                     }
                 }
             }
-        }
+        } // context - When using default storage
+
+        context("when using mock storage to analyze behavior") {
+            var mockStarage: MockStorage!
```
Comment (viniciusponciano-tw) [inline Tests/Public/DesignSystem+Spec.swift:53]: The variable name should be mockStorage instead

Comment (raonivaladares) [reply Tests/Public/DesignSystem+Spec.swift:53]: done!

**Pull request merged**